homeassistant/components/climate/__init__.py

```diff
@@ -357,7 +357,7 @@ def target_humidity(self) -> int | None:
         return self._attr_target_humidity
 
     @property
-    def hvac_mode(self) -> HVACMode | None:
+    def hvac_mode(self) -> HVACMode | str | None:
```

Comment:
We don't allow string anymore. What was the reason for the change?

Reply:
The `state` method still handles the case where the `hvac_mode` is a string:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L246

If strings are fully deprecated, I'd be happy to instead remove the above check from `state`.

Reply:
I think that's left for custom integrations. I think we can add a type ignore to the state property for now. When we decide to make the breaking change we can remove it.

Reply:
Thanks @MartinHjelmare, made those changes in the latest commit.